include/net/buf.h

```diff
@@ -537,8 +537,13 @@ struct net_buf {
 		struct net_buf_simple b;
 	};
 
-	/** System metadata for this buffer. */
-	u8_t user_data[CONFIG_NET_BUF_USER_DATA_SIZE] __net_buf_align;
+	union {
+		/** System metadata for this buffer. */
+		u8_t user_data[CONFIG_NET_BUF_USER_DATA_SIZE] __net_buf_align;
+
+		/** Allow placing network buffer into sys_slist_t */
+		sys_snode_t next;
+	};
```

Comment:
@jhedberg 
> If you want to have it in some additional list then the user_data may indeed be the right place

Thanks for the input here. At the moment there’s a need to keep it in the additional list. Eventually this will be dropped at the first such possibility/feasibility (I am considering and evaluating it).

I planned and will list this in the issues.

Reply:
@jhedberg

Done as discussed. Please take a look.

Reply:
@ozhuraki thanks! I've withdrawn my "Changes requested"